soc/riscv/riscv-privilege/nuclei/gd32vf103/soc.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file SoC configuration macros for the GigaDevice GD32V processor
+ */
+
+#ifndef __RISCV_GIGADEVICE_GD32V_SOC_H_
+#define __RISCV_GIGADEVICE_GD32V_SOC_H_
+
+#ifndef _ASMLANGUAGE
+#include <stdint.h>
+#endif
+#include <soc_common.h>
+#include <devicetree.h>
+
+/* Timer configuration */
+#define RISCV_MTIME_BASE             (0xD1000000UL + 0)
+#define RISCV_MTIMECMP_BASE          (0xD1000000UL + 0x8)
+
+/* lib-c hooks required RAM defined variables */
+#define RISCV_RAM_BASE               DT_SRAM_BASE_ADDRESS
+#define RISCV_RAM_SIZE               KB(DT_SRAM_SIZE)
```

Comment:
question: is this required to be at `soc.h`?

Reply:
RISCV_MTIME_BASE and RISCV_MTIMECMP_BASE are used by RISCV machine timer.
RISCV_RAM_BASE and RISCV_RAM_SIZE are not referenced in zephyr source.
Removing RISCV_RAM_BASE and RISCV_RAM_SIZE.